base_layer/wallet/src/text_message_service.rs

```diff
@@ -436,3 +608,69 @@ impl TextMessageServiceApi {
         res
     }
 }
+
+mod test {
+    use crate::text_message_service::{Contact, TextMessageError, TextMessageService};
+    use tari_comms::types::{CommsPublicKey, CommsSecretKey};
+    use tari_crypto::keys::{PublicKey, SecretKey};
+
+    #[test]
+    fn test_contacts_crud() {
+        let mut rng = rand::OsRng::new().unwrap();
+
+        let secret_key = CommsSecretKey::random(&mut rng);
+        let public_key = CommsPublicKey::from_secret_key(&secret_key);
```

Comment:
nit. can replace ln 621 and ln 622 with `let (secret_key, public_key) = CommsPublicKey::random_keypair(&mut rng);`

Reply:
Done